### DISCUSSION TITLE
Are there any metrics on a maximum number of subscribers?

### DISCUSSION THREAD

**Discussion by rlewkowicz** [eclipse-ecal/ecal#976] [Q&A] [answered]
Is there a maximum number of subscribers? 1k 10k 100k? 

I'm sure there's nuance to that based on payload size, I was just wondering if there's any metrics.

**Comment by KerstinKeller** [comment, marked as answer]
Very good question 😄 
We've usually used eCAL in contexts with about ~200 Pub/Sub connections max.

We now have new usecases to support scenarios of 1000-10000 Pub/Sub connections. We have identified the parts of eCAL which don't scale that well, and taking measures to make that working.
Bottleneck is the eCAL Monitoring layer (which communicates over UDP), so with eCAL 5.11 we have implemented a shared memory monitoring layer (obviously works only locally), which is performing a lot better, but still has some problems.

Another hard cap is the number of open files allowed to be open on the system (or per process, especially a problem for eCAL player), since eCAL opens a surprisingly high number (between 3 and 11) of files per publisher, to exchange the data (depends on local vs network communication etc).

I think 100k is definately out of scope for now, but who knows how well we get the system to scale 😉 

I guess it's also a question of how large a system you want to set up. Probably you don't want to have 100k processes on the same host?
And if instead you're thinking more 1 process with 100k publishers, then we'd recommend to send larger chunks of data, instead of having 1 publisher per signal.

I hope this answers your question?
